### DISCUSSION TITLE
Could rpc calls allow you to specify a path?

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#7206] [Scripting]
I'm getting into Godot 4 networking and rpc calls.  While I do like the system, one thing that irks me is that you are required to have identical objects on every participant in order to pass messages amongst each other.  While I can see how this may be useful for a game where every client is the same, it a big hassle when you're trying to create a headless server where the methods the rpc calls the client needs to call on the server are completely different from the ones the server sends back to the client.  

I'm finding myself creating a bridge object that has some methods only ever called by the client and other methods only ever called by the server.  It would be nice to break this into two separate objects.

Would it be possible to extend the rpc syntax so that you can explicitly specify the path of the remote object you're sending the call to rather than relying on every peer to have identical objects residing at identical paths?

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/7065.

- See discussion in https://github.com/godotengine/godot/issues/57869.

**Comment by blackears** [comment]
I've read through some of those threads and just don't understand the logic of forcing all rpc objects to have the same methods on them.  Is being able to address methods with integers instead of strings really saving that much time?  It also locks you into a very specific architecture.  Maybe this would work well for some game designs, but it really messes you up when different peers need to call a very different set of methods and you don't care about having any peer broadcasting the same message to every other peer.
